File: update.R
update<-function(params,grads,learning_rate){
  parameters<-params
  L<-length(parameters)%/%2
  for (l in 1:L){
    parameters[[paste0("W",l)]]<-parameters[[paste0("W",l)]]-learning_rate*grads[[paste0('dW',l)]]
    parameters[[paste0("b",l)]]<-parameters[[paste0("b",l)]]-learning_rate*grads[[paste0('db',l)]]
  }
  return(parameters)
}

File: update.jl
function update(params, grads, learning_rate)
  parameters = copy(params)
  L = div(length(parameters), 2)
  for l in 1:L
    parameters[string("W", l)] = parameters[string("W", l)] - learning_rate * grads[string("dW", l)]'
    parameters[string("b", l)] = parameters[string("b", l)] - learning_rate * grads[string("db", l)]
  end
  return parameters
end